### PR TITLE
Add prettier and integrate it with eslint

### PR DESCRIPTION
**Pull request by donegjookim**
The main goal of this PR is to integrate prettier with eslint, precommit hook, and git hub action. Besides that, I

- Added eslint check in github action for consistency
- Added React custom hooks linting rule to ensure best practices
- Added .git-blame-ignore-revs to ensure that git blame history does not get sabotaged

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (blrchen) [inline .prettierrc:1]: Please move prettierrc, .eslintignore files to under ui directory, as these files are scoped to ui code only.

Is .git-blame-ignore-revs scoped to all codes in repo? If no probably this file should get moved too.

File: .github/workflows/pull_request_push_test.yml [hunk 2]
```diff
@@ -32,6 +32,28 @@ jobs:
           distribution: "temurin"
       - name: Run tests
         run: sbt clean && sbt test
+
+  eslint:
+    name: eslint
```
Comment (blrchen) [inline .github/workflows/pull_request_push_test.yml:37]: >name: eslint [](http://example.com/codeflow?start=4&length=12)

Please add trigger condition so eslint and prettier formatter only fires when PR includes change for ui files.

File: ui/README.md [hunk 3]
```diff
@@ -47,6 +52,7 @@ REACT_APP_AZURE_TENANT_ID=<REPLACE_WITH_YOUR_TENANT_ID>
 ### Lint
```
Comment (blrchen) [inline ui/README.md:52]: Please add some contents under Lint section in README.md, so people can qucikly know what lints/code formatters have been configured. Doc should indicate where the rule are configured as well.

File: .prettierrc (hunk 1 above)
Comment (donegjookim) [reply .prettierrc]: Thanks for catching - I totally missed moving prettierrc. It's been a while since I worked with a mono repo.

Comment (donegjookim) [reply .prettierrc]: As it is related to git, .git-blame-ignore-revs is per git repo basis.
It makes sense to be at the root directory as we're only using one git repository

File: ui/README.md (hunk 3 above)
Comment (donegjookim) [reply ui/README.md]: I added comment about Prettier and mentioned that we're using the default rules.

File: .github/workflows/pull_request_push_test.yml (hunk 2 above)
Comment (donegjookim) [reply .github/workflows/pull_request_push_test.yml]: I created another file to address this issue - it'll better help to separate API checks and UI checks in the workflow for readability and maintainability

**Review by blrchen — APPROVED** [review]

**Review by Yuqing-cat — APPROVED** [review]

**Pull request merged**